lib/FileCard/styles.scss

```diff
@@ -0,0 +1,123 @@
+/* ==========================================================================
+   Config
+   ========================================================================== */
+$field-card-width: 200px !default;
+$field-card-height: 180px !default;
+$field-card-action-button-size: 32px !default;
+$field-card-margin: $layout-spacing-base !default;
+
+$field-card-label-font-size: $typo-size-slight !default;
+$field-card-actions-padding: .5rem !default;
+
+$field-card-label-bg: rgba(0, 0, 0, .7) !default;
+$field-card-highlight-color: rgb(245, 218, 116);
+/* ==========================================================================
+   Styles
+   ========================================================================== */
+.file-card {
+  width: $field-card-width;
+  height: $field-card-height;
+  margin-right: $field-card-margin;
+  position: relative;
+  display: inline-block;
+  overflow: hidden;
+
+  &:hover,
+  .file-card__action:focus {
+    @for $i from 1 through 4 {
+      .file-card__action:nth-child(#{$i}) {
+        transform: translateY(0);
+        transition-delay: #{$i/20}s;
+      }
+    }
+  }
+
+  &.has-comments {
+    border: 3px solid $field-card-highlight-color;
+    transition: border .1s ease-in;
+
+    &:hover {
+      border: 3px solid darken($field-card-highlight-color, 20%);
+    }
+  }
+}
+
+.file-card__wrapper {
+  display: flex;
+  flex-direction: column;
+}
+
+.file-card__thumbnail {
+  background-size: cover;
+  width: 100%;
+  height: 100%;
+  position: absolute;
+  overflow: hidden;
+  z-index: 0;
+
+  &:hover {
+    cursor: pointer;
+  }
+}
+
+.file-card__thumbnail--no-preview {
+  background-color: rgba(155, 155, 155, .1);
+  align-content: center;
+  justify-content: center;
+  font-size: 2em;
+  text-transform: uppercase;
+  display: flex;
+  padding-top: 35%;
+  overflow: visible;
+  font-weight: 600;
+}
+
+.file-card__label {
+  background: $field-card-label-bg;
+  color: #fff;
+  position: absolute;
+  bottom: $field-card-actions-padding;
+  font-size: $field-card-label-font-size;
+  border-radius: 3px;
+  padding: .2rem $field-card-actions-padding;
+  max-width: 90%;
+  text-overflow: ellipsis;
+  overflow: hidden;
+  white-space: nowrap;
+  align-self: center;
+  text-align: center;
+}
+
+.file-card__actions {
+  display: flex;
+  padding: .5rem;
+  justify-content: flex-start;
+
+  > .file-card__action {
+    @for $i from 1 through 4 {
+      &:nth-child(#{$i}) {
+        transform: translateY(-3rem);
```

Comment:
I'll probably take care of this once (and if!) @nhqe approves the animation at all to begin with, if that's alright! No point in optimising it now if it will be taken away in the end :)